pwiz_tools/Skyline/TestUtil/TestFunctional.cs

```diff
@@ -2439,8 +2439,11 @@ public static void SetStaticModifications(Func<IList<string>, IList<string>> cha
             if (expectedErrorMessage != null)
             {
                 var dlg = WaitForOpenForm<MessageDlg>();
-                Assert.IsTrue(dlg.DetailMessage.Contains(expectedErrorMessage));
-                dlg.CancelButton.PerformClick();
+                RunUI(() =>
```

Comment:
I think it would be good to make this a RunDlg() or use OkDialog() to close the form just in case the next instruction tries to do something that shows a MessageDlg. This code doesn't ensure that the MessageDlg is gone before continuing on.